examples/out-of-tree/cloud-controller-manager.yaml

```diff
@@ -129,68 +129,77 @@ subjects:
     kind: User
     name: cloud-controller-manager
 ---
-apiVersion: v1
-kind: Pod
+apiVersion: apps/v1
+kind: Deployment
 metadata:
   name: cloud-controller-manager
   namespace: kube-system
   labels:
-    tier: control-plane
     component: cloud-controller-manager
 spec:
-  priorityClassName: system-node-critical
-  hostNetwork: true
-  serviceAccountName: cloud-controller-manager
-  tolerations:
-    - key: node-role.kubernetes.io/master
-      effect: NoSchedule
-  containers:
-    - name: cloud-controller-manager
-      image: mcr.microsoft.com/oss/kubernetes/azure-cloud-controller-manager:v1.23.5
-      imagePullPolicy: IfNotPresent
-      command: ["cloud-controller-manager"]
-      args:
-        - "--allocate-node-cidrs=true" # "false" for Azure CNI and "true" for other network plugins
-        - "--cloud-config=/etc/kubernetes/azure.json"
-        - "--cloud-provider=azure"
-        - "--cluster-cidr=10.244.0.0/16"
-        - "--cluster-name=k8s"
-        - "--controllers=*,-cloud-node" # disable cloud-node controller
-        - "--configure-cloud-routes=true" # "false" for Azure CNI and "true" for other network plugins
-        - "--leader-elect=true"
-        - "--route-reconciliation-period=10s"
-        - "--v=2"
-        - "--port=10267"
-      resources:
-        requests:
-          cpu: 100m
-          memory: 128Mi
-        limits:
-          cpu: "4"
-          memory: 2Gi
-      livenessProbe:
-        httpGet:
-          path: /healthz
-          port: 10267
-        initialDelaySeconds: 20
-        periodSeconds: 10
-        timeoutSeconds: 5
-      volumeMounts:
+  replicas: 1
```

Comment:
change to 2 for HA?

Reply:
Updated.

Reply:
@feiskyer I manually tested the new yaml file. Then it seems that on a cluster with one control-plane node, 2 ccm are not allowed.

```
kube-system                         cloud-controller-manager-86c84c49b7-g9gth                       1/1     Running            1          2m19s   172.18.0.3    capz-control-plane   <none>           <none>
kube-system                         cloud-controller-manager-86c84c49b7-gzcbp                       0/1     CrashLoopBackOff   4          2m19s   172.18.0.3    capz-control-plane   <none>           <none>
```

The failed one Pod's log:
```
E0316 03:32:08.120288       1 controllermanager.go:78] Run: failed to configure cloud controller manager: failed to create listener: failed to listen on 0.0.0.0:10267: listen tcp 0.0.0.0:10267: bind: address already in use
```

Reply:
OK, then let's keep 1 here